src/reqstool/model_generators/requirements_model_generator.py

```diff
@@ -302,6 +303,13 @@ def __parse_requirements(self, data):  # NOSONAR
                 rationale = req["rationale"] if "rationale" in req else None
                 # Check if implementation is defined, or set it to True
                 implementation = req["implementation"] if "implementation" in req else IMPLEMENTATION.IN_CODE.value
+                # Get lifecycle variables or use defaults
+                if "lifecycle" in req:
+                    lifecycle_state = LIFECYCLESTATE(req["lifecycle"]["state"])
+                    lifecycle_reason = req["lifecycle"]["reason"] if "reason" in req["lifecycle"] else ""
```

Comment:
Don't use empty string for missing values in general, so if there is no reason in the input then use None instead .

The user could actually use the empty string, but that's a different story.